src/globus_sdk/experimental/globus_app/app.py

```diff
@@ -84,22 +94,23 @@ def __init__(
             config=self.config,
         )
 
-        # construct ValidatingTokenStorage around the TokenStorage and
-        # our initial scope requirements
-        self.token_storage = ValidatingTokenStorage(
+        # create a consent client for token validation
+        # reducing the scope requirements to barebones openid (user identification)
+        # additionally, this will ensure that openid scope requirement is always
+        # registered (it's required for token identity validation).
+        consent_client = AuthClient(app=self, app_scopes=[Scope(AuthScopes.openid)])
+
+        # create the requisite token storage for the app, with validation based on
+        # the provided parameters
+        self.token_storage = _build_default_validating_token_storage(
             token_storage=self._token_storage,
+            consent_client=consent_client,
             scope_requirements=self._scope_requirements,
         )
 
         # initialize our authorizer factory
         self._initialize_authorizer_factory()
-
-        # create a consent client for token validation
-        # reducing the scope requirements to barebones openid (user identification)
-        # additionally, this will ensure that openid scope requirement is always
-        # registered (it's required for token identity validation).
-        consent_client = AuthClient(app=self, app_scopes=[Scope(AuthScopes.openid)])
-        self.token_storage.set_consent_client(consent_client)
+        self._authorizer_factory_initialized = True
```

Comment:
I don't think there's any perfect solution for this -- it's an attempt to capture the "partially initialized state" -- and I don't mind adjusting it.

We could definitely set it to False at the start of init, rather than as a class attr. I'd be fine with that.

If we don't set it to True at the end of init, each subclass will be required to set it in their implementations of the `_initialize_authorizer_factory` hook. I think that's not great, and risks having some subclass miss it. So I'd rather keep that behavior in the base class if possible.

Another solution for this whole problem is to implement a class `_InitStubAuthorizerFactory` which gets assigned at the start of init, and overwritten later. It can raise errors (except for the place where this bool is checked, which would be a no-op. We can then eliminate the separate bool. I only just thought of this idea, but let me know if it sounds worth pursuing.

Reply:
I lean toward setting it to `False` at the start and setting it to `True` at the end. :+1:

Reply:
I wonder if it'd read cleaner if the flag was just `_initialized` rather than being specific to the authorizer.

---

I hadn't considered the fake authorizer factory route. That would clean this up pretty well...

Reply:
I tried it both ways and settled on the bool. I've pushed the stub-object version followed by a revert commit so that we can all see the two side-by-side. I think the simplicity of the bool wins out by a lot, when you write the whole thing out. Having to provide the stub with all of the methods of the ABC is one of the issues with that approach, IMO.